src/libcore/num/mod.rs

```diff
@@ -321,6 +321,33 @@ $EndFeature, "
             (self as $UnsignedT).swap_bytes() as Self
         }
 
+        /// Reverses the bit pattern of the integer.
+        ///
+        /// # Examples
+        ///
+        /// Please note that this example is shared between integer types.
+        /// Which explains why `i16` is used here.
```

Comment:
@GuillaumeGomez @QuietMisdreavus I feel like I've recently heard something about us being able to generate dedicated examples with the right types for primitives...

Reply:
Yep we did [here](https://github.com/rust-lang/rust/pull/48152).

Reply:
I just copied the docs for `swap_bytes` and made minor adjustments.

Reply:
Well, in this case it is very complicated to be able to write a fully generic example.